### DISCUSSION TITLE
How to model multiple table inheritance?

### DISCUSSION THREAD

**Discussion by Sleepful** [pocketbase/pocketbase#682] [Q&A]
Single table inheritance can be done rather easily but I am not sure on how to model multiple table inheritance.

I want to create a `category_collection` which will have a `category_name` field and a `related_collection` field, the `related_collection` might point to one of many collections, all of the collections will have a `value` field. I want to be able to query the `category_collection` to find a record by `category_name` and find the related value through `related_collection`, the value could be different types. 

Currently I am going to attempt making `related_collection` a text field with the name of the collection, and try to query that... but I assume that this will be poor performance/API, perhaps I will need to use `PocketBase` as a Go framework?

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand the described data structure.

Are you asking for doing something like:
```js
// we use getList with 1 item limit because we currently don't have a helper to query a single record with custom filter 
const categories = await client.records.getList('category_collection', 1, 1, {
    filter: 'category_name = "YOUR_CATEGORY_NAME"',
});
const category = categories.items[0];

// fetch the related collection records by the `value` field
const relatedRecords = await client.records.getList(category.related_collection, 1, 50, {
    filter: 'value = "some value"',
});
```
?

I wouldn't worry too much about the performance, if that is your concern.

If the above is not what you are looking for, could you elaborate a little more on your use case (an example erd would help)?

**Reply by Sleepful** [reply]
Yes this is what I meant, in your example the client (sdk consumer) is creating two separate API calls, I believe ideally the joins would be performed at the SQLite layer. For 1 record this isn't a big deal, but if I am trying to fetch some 50 records then this becomes 50 API calls. Unless I start to optimize the queries in the client by trying to batch them together, which gets messy.

Perhaps I am worrying too much and, like you said, I should not worry about this. I am at the design stage after all, I haven't coded this yet.

IIRC [Ruby's Active Record](https://user3141592.medium.com/when-to-use-single-table-inheritance-vs-multiple-table-inheritance-db7e9733ae2e) and [Python's SqlAlchemy](https://docs.sqlalchemy.org/en/14/orm/inheritance.html) have ways of doing this.

**Reply by ganigeorgiev** [reply]
I'm not sure if what you want could be achieved in a single select with a join, because the joined table name in your case is dynamic and depends on the resolved column value, aka. I don't think you can do directly `LEFT JOIN category_collection.related_collection` and most likely you'll have to join the `related_collection` column as subquery or something similar.

**Reply by Sleepful** [reply]
That's a good point, thanks for your input, let me look into this a little longer before coming back with an answer
